design/baremetal-operator/external-introspection.md

```diff
@@ -0,0 +1,116 @@
+# Support external introspection
+
+## Status
+
+implementable
+
+## Summary
+
+A declarative API is proposed to disable inspection of a BareMetalHost
+and optionally allow external sources of inspection data to update the
+hardware status data.
+
+## Motivation
+
+Related to the work to enable booting of a
+[LiveImage](https://github.com/metal3-io/metal3-docs/pull/150),
+there is the requirement to optionally disable inspection on initial
+registration of a BareMetalHost (so that the live image can boot
+more quickly, ref user stories below).
```

Comment:
@hardys Do you know if there's any other scenario, rather than the live-iso one, requiring to disable the initial inspection? If it's the only one, couldn't be maybe enough to detect the presence of `spec.image.format == live-iso` as described in the [BMH live-iso proposal](https://github.com/metal3-io/metal3-docs/blob/master/design/baremetal-operator/bmh_live_iso.md) to skip the inspection?

Reply:
@andfasano I don't think such an option is specific to live-iso deployments, this could be used any time deploy walltime is desired above collecting hardware details?  

Particularly for some repeated CI/test cases on VMs I can imagine disabling inspection to reduce the time for test iterations.

Reply:
Thanks, just wanted to understand if there was any known use case.

Reply:
> Particularly for some repeated CI/test cases on VMs I can imagine disabling inspection to reduce the time for test iterations.

If this is the reason, let's not do this. This is a dangerous path to take which will necessarily lead to broken inspection in the end.

Reply:
Perhaps walltime optimization was a bad example to use, and I agree we probably shouldn't (ab)use this sort of interface in CI, my point was really that I don't think we should couple the `inspect.metal3.io` annotation to `spec.image.format == live-iso `, since they're essentially two orthogonal interfaces.

As @dhellmann mentions above it's the responsibility of the user to consider the impact of any API use, and provided inspection is enabled by default, and we clearly document the ramifications of disabling it, I don't see much benefit in implementing hidden dependencies between two essentially independent BMH APIs?

Reply:
I agree with keeping the APIs separate. Our current use case for the new feature has to do with live ISOs, but that's not the only one (we've discussed in another thread on this PR using it in the openshift installer instead of the current method of loading the entire status block).